src/config/watcher.rs

```diff
@@ -19,21 +19,60 @@ const CONFIG_WATCH_DELAY: std::time::Duration = std::time::Duration::from_secs(1
 
 const RETRY_TIMEOUT: std::time::Duration = std::time::Duration::from_secs(10);
 
+pub enum ConfigWatcher {
```

Comment:
I found confusing that this PR includes two `ConfigWatcher` struct with the same name but different meanings

Reply:
yeah , one i used  enum for clap options and another one in watcher.

if u can suggest me a different name. i will change